valohai_yaml/objs/task.py

```diff
@@ -0,0 +1,109 @@
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.objs.utils import (
+    check_type_and_listify,
+    consume_array_of,
+)
+
+
+class TaskParameterType(Enum):
+    """Represents a task parameter style definition."""
```

Comment:
Misnamed - this doesn't have much to do with parameters. 

```suggestion
class TaskType(Enum):
    """Represents a task type."""
```